### PR TITLE
Switch guessers to catching and raising NoDataErrors

### LINKED ISSUE #4749: Guesser checks should wrap NoDataError
Related to #4748 

In the new guesser API, we often do a check for an attribute and then except on an AttributeError. This makes for an ugly capture for users when a NoDataError gets thrown instead.

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #4749 (maybe?)

Changes made in this Pull Request:
 - Switches errors to NoDataErrors

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4755.org.readthedocs.build/en/4755/

### CODE REVIEW

**Review by orbeckst — APPROVED** [review]
From reading the issue and glancing at the code, changing the raised ValueError to NoDataError makes sense and seems more consistent.

I am not really familiar with the new guesser code so this is just a surface-level review. Other reviews would be great.

@lilyminium I'll leave it to you to gather more opinions if you need them.

**Comment by lilyminium** [comment]
@IAlibay could you please have a look to see if this resolves issue #4749?

**Comment by yuxuanzhuang** [comment]
I think what @IAlibay means is that there is a nested exception error message:

```python
a = np.array([
    [0., 0., 150.], [0., 0., 150.], [200., 0., 150.],
    [0., 0., 150.], [100., 100., 150.], [200., 100., 150.],
    [0., 200., 150.], [100., 200., 150.], [200., 200., 150.]
])
u = mda.Universe(a, n_atoms=9, to_guess=['elements'])

> ...
> NoDataError: This Universe does not contain name information
> During handling of the above exception, another exception occurred:
> ...
> NoDataError: there is no reference attributes in this universeto guess types from

```
which is kinda ugly. you can explicitly suppress the chaining of exceptions using the from None syntax (see suggestions)

File: package/MDAnalysis/guesser/default_guesser.py [hunk 1]
```diff
@@ -218,15 +219,15 @@ def guess_masses(self, atom_types=None, indices_to_guess=None):
...
                     try:
                         atom_types = self.guess_types(
                             atom_types=self._universe.atoms.names)
-                    except ValueError:
-                        raise ValueError(
+                    except NoDataError:
+                        raise NoDataError(
                             "there is no reference attributes"
                             " (elements, types, or names)"
                             " in this universe to guess mass from")
```
Comment (yuxuanzhuang) [inline package/MDAnalysis/guesser/default_guesser.py]:
```suggestion
                            " in this universe to guess mass from") from None
```

File: package/MDAnalysis/guesser/default_guesser.py [hunk 2]
```diff
@@ -291,8 +292,8 @@ def guess_types(self, atom_types=None, indices_to_guess=None):
         if atom_types is None:
             try:
                 atom_types = self._universe.atoms.names
-            except AttributeError:
-                raise ValueError(
+            except NoDataError:
+                raise NoDataError(
                     "there is no reference attributes in this universe"
                     "to guess types from")
```
Comment (yuxuanzhuang) [inline package/MDAnalysis/guesser/default_guesser.py]:
```suggestion
                    "to guess types from") from None
```

**Review by yuxuanzhuang — DISMISSED** [review]

**Comment by IAlibay** [comment]
> I think what @IAlibay means is that there is a nested exception error message:
> 
> which is kinda ugly. you can explicitly suppress the chaining of exceptions using the from None syntax (see suggestions)

Yes, thanks so much @yuxuanzhuang, that's exactly what I was looking at! Sorry I conflated that with the error type for some reason.

**Review by IAlibay — APPROVED** [review]
LGTM with @yuxuanzhuang's suggestions.

**Pull request merged**